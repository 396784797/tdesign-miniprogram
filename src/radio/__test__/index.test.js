import path from 'path';
import simulate from 'miniprogram-simulate';

describe('radio', () => {
  // test('comp should trigger change events', async () => {
  //   const container = simulate.render(radio);
  //   const comp = container.instance;
  //   comp.triggerEvent = jest.fn();
  //   comp.doChange();
  //   expect(comp.triggerEvent).toHaveBeenCalled();
  // });
  // const id = simulate.load(path.resolve(__dirname, `./index`), { less: true });

  describe('with group', () => {
    it(`radio :base`, async () => {
      const id = simulate.load({
        template: `
        <t-radio-group class="group" value="1">
          <t-radio class="a" value="1"></t-radio>
        </t-radio-group>
        `,
        usingComponents: {
          't-radio': './radio',
          't-radio-group': '../../radio-group/radio-group',
        },
        rootPath: path.resolve(__dirname, '../..'),
        compiler: 'official',
      });
      const comp = simulate.render(id);
      comp.attach(document.createElement('parent-wrapper'));

      // const group = comp.querySelector('.group')
      const a = comp.querySelector('.a');

      expect(a.data.checked).toBeTruthy();
      // expect(comp.toJSON()).toMatchSnapshot();
    });
  });
});
