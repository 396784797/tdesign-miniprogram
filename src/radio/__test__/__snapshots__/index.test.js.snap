// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`radio with group radio :base 1`] = `
<main>
  <t-radio-group
    class="group"
  >
    <wx-view
      class="t-radio-group t-class"
    >
      <t-radio
        class="a"
      >
        <wx-view
          class="t-radio t-class t-radio--left"
        >
          <wx-view
            class="t-radio__icon t-radio__icon--left t-class-icon"
            data-target="icon"
            bind:tap="handleTap"
          >
            <t-icon
              class="t-radio__btn  "
            >
              <wx-label
                class="t-class t-icon t-icon-circle t-icon-base"
                style=";;"
                bind:tap="onTap"
              />
            </t-icon>
          </wx-view>
          <wx-view
            class="t-radio__content "
            data-target="text"
            bind:tap="handleTap"
          >
            <wx-view
              class="t-radio__title t-class-label"
              style="-webkit-line-clamp:3"
            >
              
      
      
            </wx-view>
            <wx-view
              class="t-radio__description t-class-content "
              style="-webkit-line-clamp:5"
            >
              
      
    
            </wx-view>
          </wx-view>
          <wx-view
            class="t-radio__border t-radio__border--left t-class-border"
          />
        </wx-view>
      </t-radio>
    </wx-view>
  </t-radio-group>
</main>
`;
